### PR TITLE
introducing kafka-lagcheck

### PR DESCRIPTION
**Pull request by kovacshuni**
https://github.com/Financial-Times/kafka-lagcheck
https://github.com/Financial-Times/burrow
https://github.com/Financial-Times/pub-service-files/pull/55

* Allowed lag is set to 30 currently.
* Concept is an ignored partition.
* Burrow can show partitions in unhealthy state even if lag is less than 30, if it is seeing the stalled or stopped. (https://github.com/linkedin/Burrow/wiki/Consumer-Lag-Evaluation-Rules) These are special cases, we'll need to see this lagcheck running in live, and see its behaviour during a mass publish.
* Can be acked any time.
* Memory limit on both apps: 256MB
* Present in xp, pub-xp, pub-pre-prod-uk, pre-prod-uk

### CODE REVIEW

File: kafka-lagcheck@.service [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ExecStartPre=-/bin/bash -c '/usr/bin/docker rm "$(docker ps -q --filter=name=%p-%i_)" > /dev/null 2>&1'
+ExecStartPre=/bin/bash -c '/usr/bin/docker history coco/kafka-lagcheck:$DOCKER_APP_VERSION > /dev/null 2>&1 \
+  || /usr/bin/docker pull coco/kafka-lagcheck:$DOCKER_APP_VERSION'
+  
+ExecStart=/bin/sh -c '\
+  /usr/bin/docker run --rm --name %p-%i_$(uuidgen) -p 8080 \
+    --memory="256m" \
+    --env "HOST_MACHINE=%H" \
+    --env "WHITELISTED_TOPICS=Concept" \
+    --env "LAG_TOLERANCE=30" \
```
Comment (tamas-molnar) [inline kafka-lagcheck@.service]: Can you read this from ETCD please? It can be more easily adjusted.

File: burrow-sidekick@.service [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+BindsTo=burrow@%i.service
+After=burrow@%i.service
+
+[Service]
+RemainAfterExit=yes
+ExecStart=/bin/sh -c "\
+  export SERVICE=$(echo %p | sed 's/-sidekick//g'); \
+  etcdctl set /ft/services/$SERVICE/healthcheck false; \
+  etcdctl mkdir /ft/services/burrow/servers; \
+  while [ -z $BURROW_PORT ]; do \
```
Comment (tamas-molnar) [inline burrow-sidekick@.service]: `$PORT` is enough here, see https://github.com/Financial-Times/up-service-files/blob/master/template/template-sidekick%40.service 
Keeps the sidekicks consistent.

File: kafka-lagcheck-sidekick@.service [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+After=kafka-lagcheck@%i.service
+
+[Service]
+RemainAfterExit=yes
+ExecStart=/bin/sh -c "\
+  export SERVICE=$(echo %p | sed 's/-sidekick//g'); \
+  etcdctl set /ft/services/$SERVICE/healthcheck true; \
+  etcdctl set /ft/healthcheck/$SERVICE-%i/path /__health; \
+  etcdctl mkdir /ft/services/$SERVICE/servers; \
+  while [ -z $KAFKA_LAGCHECK_PORT ]; do \
```
Comment (tamas-molnar) [inline kafka-lagcheck-sidekick@.service]: `$PORT` is enough here, see https://github.com/Financial-Times/up-service-files/blob/master/template/template-sidekick%40.service 
Keeps the sidekicks consistent.

File: kafka-lagcheck-sidekick@.service [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    sleep 5; \
+    CONTAINER_NAME=$(docker ps -q --filter=name=\"$SERVICE\"-%i_); \
+    KAFKA_LAGCHECK_PORT=`echo $(/usr/bin/docker port $CONTAINER_NAME 8080 | cut -d':' -f2)`; \
+  done; \
+  etcdctl set /ft/services/$SERVICE/servers/%i http://%H:$KAFKA_LAGCHECK_PORT;"
+
+ExecStop=-/bin/bash -c '/usr/bin/etcdctl rm /ft/services/kafka-lagcheck/servers/%i \
+  /usr/bin/etcdctl rm /ft/services/burrow/servers/%i'
```
Comment (tosan88) [inline kafka-lagcheck-sidekick@.service]: this is already been removed in `burrow-sidekick@.service`

File: kafka-lagcheck@.service (hunk 1 above)
Comment (kovacshuni) [inline kafka-lagcheck@.service]: done

File: burrow-sidekick@.service (hunk 2 above)
Comment (kovacshuni) [inline burrow-sidekick@.service]: done

File: kafka-lagcheck-sidekick@.service (hunk 3 above)
Comment (kovacshuni) [inline kafka-lagcheck-sidekick@.service]: done

File: kafka-lagcheck-sidekick@.service (hunk 4 above)
Comment (kovacshuni) [inline kafka-lagcheck-sidekick@.service]: fixed

**Pull request merged**